nav2_controller/src/nav2_controller.cpp

```diff
@@ -94,6 +96,11 @@ ControllerServer::on_configure(const rclcpp_lifecycle::State & state)
 
   get_parameter("controller_frequency", controller_frequency_);
   RCLCPP_INFO(get_logger(), "Controller frequency set to %.4fHz", controller_frequency_);
+  callback_handles_.push_back(param_subscriber_->add_parameter_callback("controller_frequency",
+    [&](const rclcpp::Parameter & p) {
+      std::lock_guard<std::recursive_mutex> lock(mutex_);
+      controller_frequency_ = p.get_value<double>();
```

Comment:
This will only update the controller frequency after navigation is complete. We need to take explicit action in the main loop to update the rate object if this changes.

Reply:
Right, I noticed that.

Reply:
Ok. I guess the question is, "is it ok like it is?". I'm willing to go with it for now until someone asks for more.